File: ..solutions/53. Maximum Subarray.cpp
using arr = vector<int>;
const int inf = 2e9+18;
int maxx = -1 * inf;
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size() ;
        arr dp(n);
        dp[0] = nums[0];
        maxx = dp[0];
        for(int i = 1; i < n; i++){
            if(dp[i-1] + nums[i] > nums[i]){
                dp[i] = dp[i-1] + nums[i];
            }else{
                dp[i] = nums[i];
            }
            maxx = max(dp[i],maxx);
        }
        return maxx;
    }
};

File: ..solutions/53. Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] arr) {
        int n = arr.length;
        int maximumsum = arr[0];
        int runningSum = arr[0];
​
        for (int i = 1; i < n; i++) {
            runningSum += arr[i];
            if (runningSum > arr[i]) {
                maximumsum = Math.max(maximumsum, runningSum);  
            } else {
                runningSum = arr[i];
                maximumsum = Math.max(maximumsum, runningSum);
            }
        }
​
        return maximumsum;
    }
}
